kits/token-wrapper/contracts/TokenWrapper.sol

```diff
@@ -0,0 +1,143 @@
+    string private constant ERROR_TOKEN_BURN_FAILED = "TW_TOKEN_BURN_FAILED";
+    string private constant ERROR_TOKEN_MINT_FAILED = "TW_TOKEN_MINT_FAILED";
+    string private constant ERROR_ERC20_TRANSFER_FAILED = "TW_ERC20_TRANSFER_FAILED";
+    string private constant ERROR_ERC20_TRANSFER_FROM_FAILED = "TW_ERC20_TRANSFER_FROM_FAILED";
+
+    bytes32 public constant CONTROLLER_MANAGER_ROLE = keccak256("CONTROLLER_MANAGER_ROLE");
+
+    ERC20 public erc20;
+    MiniMeToken public token;
+    mapping(address => uint256) internal lockedAmount;
+
+    event TokensLocked(address entity, uint256 amount);
+    event TokensUnlocked(address entity, uint256 amount);
+
+    modifier onlyToken() {
+        require(msg.sender == address(token), ERROR_CALLER_NOT_TOKEN);
+        _;
+    }
+
+    /**
+     * @notice Initialize a new MiniMe token that will be the controller of an ERC20 token
+     * @param _token The MiniMe token that is used in the DAO
+     * @param _erc20 The ERC20 token that is locked in order to receive MiniMe tokens
+     */
+    function initialize(MiniMeToken _token, ERC20 _erc20) external {
+        initialized();
+        require(_token.controller() == address(this), ERROR_INVALID_TOKEN_CONTROLLER);
+
+        token = _token;
+        erc20 = _erc20;
+        token.enableTransfers(false);
+    }
+
+    /**
+     * @notice Lock `_amount` tokens
+     */
+    function lock(uint256 _amount) external {
+        require(_amount > 0, ERROR_LOCK_AMOUNT_ZERO);
+
+        lockedAmount[msg.sender] = lockedAmount[msg.sender].add(_amount);
+        emit TokensLocked(msg.sender, _amount);
+
+        require(erc20.transferFrom(msg.sender, address(this), _amount), ERROR_ERC20_TRANSFER_FROM_FAILED);
+        require(token.generateTokens(msg.sender, _amount), ERROR_TOKEN_MINT_FAILED);
+    }
+
+    /**
+     * @notice Unlock `_amount` tokens
+     */
+    function unlock(uint256 _amount) external {
+        require(_amount <= lockedAmount[msg.sender], ERROR_INVALID_UNLOCK_AMOUNT);
+
+        lockedAmount[msg.sender] = lockedAmount[msg.sender].sub(_amount);
+        emit TokensUnlocked(msg.sender, _amount);
+
+        require(token.destroyTokens(msg.sender, _amount), ERROR_TOKEN_BURN_FAILED);
+        require(erc20.transfer(msg.sender, _amount), ERROR_ERC20_TRANSFER_FAILED);
+    }
+
+    /**
+     * @notice Give control of the MiniMe token over to `_newController`
+     * @param _newController Ethereum address of contract that will control the token
+     */
+    function changeTokenController(address _newController) external auth(CONTROLLER_MANAGER_ROLE) {
+        token.changeController(_newController);
+    }
+
+    /*
+    * @dev Notifies the controller about a token transfer allowing the controller to react if desired
+    * @return False aways
+    */
+    function onTransfer(address, address, uint256) external onlyToken returns (bool) {
+        return false;
```

Comment:
I'm not fully aware of the purpose of this wrapper contract. 
What's the reason for not allowing transfers and approvals? Is it to keep the locking accountancy simpler?
Should somebody who wants to transfer first unlock, transfer the ERC20 tokens and then have the recipient to lock the received ones? Looks like quite inconvenient.

Reply:
Yea, currently holders will have to first unlock in order to transfer their tokens. I think it makes it simpler to deal with transfer so it doesn't have to update the locked values, but I don't see an issue in case we want to support users to allow or transfer tokens.